### PR TITLE
chore: Update DataFusion pin

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
* Keep up with dependencies
* Part of https://github.com/influxdata/idpe/issues/18117

# Changes:
1. Update DataFusion pin to https://github.com/apache/arrow-datafusion/commit/fda561e803d309018368696608afced4e771d17e

### CODE REVIEW

File: test_helpers_end_to_end/src/error.rs [hunk 1]
```diff
@@ -26,8 +26,8 @@ pub fn check_tonic_status(
     );
     if let Some(expected_message) = expected_message {
         let status_message = status.message();
-        assert_eq!(
-            status_message, expected_message,
+        assert!(
```
Comment (alamb) [inline test_helpers_end_to_end/src/error.rs:29]: this is needed as now the error messages contain backtraces: https://github.com/apache/arrow-datafusion/pull/7434

I do wonder if we might want to slim them down (as in not pass along the stack traces) but for now I think more information the better.

Comment (crepererum) [reply test_helpers_end_to_end/src/error.rs:29]: I'm not sure if including backtraces in error messages is a good idea at all. I think it's a hack, stating that you messed up error handling so much that you cannot even make sense of your own error type :shrug:

**Review by crepererum — APPROVED** [review]

File: iox_query/src/logical_optimizer/influx_regex_to_datafusion_regex.rs [hunk 2]
```diff
@@ -67,7 +66,7 @@ fn optimize(plan: &LogicalPlan) -> Result<LogicalPlan, DataFusionError> {
         .map(|expr| rewrite_preserving_name(expr, &mut expr_rewriter))
         .collect::<Result<Vec<_>, DataFusionError>>()?;
 
-    from_plan(plan, new_exprs.as_slice(), new_inputs.as_slice())
+    plan.with_new_exprs(new_exprs, &new_inputs)
```
[the remark this answers is not in the document]
Comment (alamb) [reply iox_query/src/logical_optimizer/influx_regex_to_datafusion_regex.rs:69]: Yeah, I deprecated it as part of https://github.com/apache/arrow-datafusion/pull/7396 (which we found when someone tried to document more nicely how to write an optimizer pass). I think the new interface is easier to discover and more consistent with the rest of the plan manipulation code

**Comment by alamb** [comment]
This introduced a significant planning time overhead regression, as @crepererum noted here https://github.com/apache/arrow-datafusion/issues/7522

I will revert the upgrade as we sort it out upstream

**Pull request merged**